src/Clide/Adapters/VsToSolutionAdapter.cs

```diff
@@ -28,16 +28,18 @@ internal class VsToSolutionAdapter :
         }
 
         public IProjectNode Adapt(IVsHierarchy from) =>
-            nodeFactory
-                .Value
-                .CreateNode(GetHierarchyItem(from))
-                as IProjectNode;
+            from is FlavoredProject && from.TryGetInnerHierarchy(out var innerHierarchy) ?
+                Adapt(new FlavoredProject(from, innerHierarchy)) :
+                nodeFactory
+                    .Value
+                    .CreateNode(GetHierarchyItem(from))
+                    as IProjectNode;
 
         public IProjectNode Adapt(FlavoredProject from) =>
             (nodeFactory
                 .Value
-                .CreateNode(GetHierarchyItem(from.Hierarchy))
-                as ProjectNode).WithInnerHierarchy(GetHierarchyItem(from.InnerHierarchy));
+                .CreateNode(GetHierarchyItem(from.InnerHierarchy))
+                as ProjectNode).WithFlavorHierarchy(from.Hierarchy);
```

Comment:
This could be improved by not assuming the result of the `CreateNode` will always be `ProjectNode`, but currently, that is indeed the case.